frame/staking-rewards/README.md

```diff
@@ -56,18 +59,82 @@ Assuming the penalty is defined as being `0.5`, if I staked `10_000
 PICA`, unstaking the NFT will result in a penalty of  `5_000 PICA`. All
 harvested rewards are still in possession of the user.
 
-## Unstaking once the NFT expired
+### Unstaking once the NFT expired
 
 In this canonical case, the user is able to unstake and rapatriate it's staked
 asset with no penalty.
 
-## Claiming
+### Claiming
 
 A user is able to claim his pending rewards at any point in time.
 As the protocol is rewarding the NFT itself, all pending rewards are tied to the NFT only.
 Meaning that if a user trade a NFT including pending rewards, the new owner will be able to unlock them.
 
-## Rewarding stakers
+### Rewarding stakers
 
 Any protocol is able to rewards the stakers by calling the
 `StakingRewards::transfer_reward` implementation of this pallet.
+
+### Epoches
+
+Rewards accumulated in epochs.
+
+User may stake amount and get fNFT, but it will not start earn until next epoch start.
+
+After each epoch end, there is short period of time it takes to collect rewards and put new fNFTs into rewarding state.
+
+During epoch user are protected from dilution made by other users which stake.
+
+### Positions operations
+
+
+
+#### Split positions
+
+Allows to split positons without paying penalty.
+
+If user had fNFT with MONTH lock total, with 2 weeks already passed with 100 tokens.
+
+It can split fNFT into several parts, examples 20, 30, 40 tokens. Each of which will be same lock duraion and time lock passed.
+
+Splitting will not loose rewards for ongoing epoch.
+
+Can be made only during epoch, but not during epoch transitions (for scalability and corretness reasons).
+
+#### Extend positions
```

Comment:
This interface yet to be defined?

Reply:
Also will it allow reduction?

Reply:
It might make sense to make the method name `resize` and allow sizing whichever way the caller wants by providing size. It's about this https://github.com/ComposableFi/composable/blob/68ae6813c20b58be4bf81ed20fadcd2adcc27d3f/rfcs/0004-pablo-distribution.md#5223-lp-fnft-resizing-or-unstaking-when-burning-lp-tokens-for-removing-liquidity

Reply:
so to reduce, split and unstake/claim
to extend, there are extend add amount operations

Reply:
as on meeting i think